urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/jasl/utilities/Serialization.java

```diff
@@ -0,0 +1,140 @@
+
+	// Public static methods
+
+	/**
+	 * Write the specified object to the indicated file.
+	 *
+	 * @param object the object to be serialized.
+	 * @param filename the name of the file that the serialized data
+	 * should be written to.
+	 *
+	 * @throws NullPointerException in the case of a null object or file name
+	 * @throws IllegalArgumentException in the case of a zero length file
+	 * name
+	 */
+
+	public static final void serializeToFile(Object object,String filename)
+	{
+		// Define local constants.
+
+		String METHOD_NAME = "serializeToFile";
+
+		// Check the parameters received and throw the appropriate
+		// exception if necessary.
+
+		if ((null == object) || (null == filename))
+		{
+			throw new NullPointerException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                          METHOD_NAME,
+			                                                          Messages.NULL_PARAMETER_MSG));
+		}
+
+		if (0 == filename.length())
+		{
+			throw new IllegalArgumentException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                              METHOD_NAME,
+			                                                              Messages.ZERO_LENGTH_PARAMETER_MSG));
+		}
+
+		try
+		{
+			ObjectOutputStream objectOutputStream =
+				new ObjectOutputStream(new FileOutputStream(filename));
+			objectOutputStream.writeObject(object);
+			objectOutputStream.close();
+		}
+
+		catch (IOException exception)
+		{
+			System.out.println("Caught: " + exception);
+		}
+	}
+
+	/**
+	 * Deserialize the data in the specified file into an object.
+	 *
+	 * @param filename the name of the file containing the data to be
+	 * deserialized.
+	 *
+	 * @return the <CODE>Object</CODE> created during deserialization of the
+	 * file data. If the process fails, the return value is null.
+	 *
+	 * @throws NullPointerException in the case of a null file name
+	 * @throws IllegalArgumentException in the case of a zero length file
+	 * name
+	 */
+
+	public static final Object deserializeFromFile(String filename)
+	{
+		Object object = null;
+
+		// Define local constants.
+
+		String METHOD_NAME = "deserializeFromFile";
+
+		// Check the parameters received and throw the appropriate
+		// exception if necessary.
+
+		if (null == filename)
+		{
+			throw new NullPointerException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                          METHOD_NAME,
+			                                                          Messages.NULL_PARAMETER_MSG));
+		}
+
+		if (0 == filename.length())
+		{
+			throw new IllegalArgumentException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                              METHOD_NAME,
+			                                                              Messages.ZERO_LENGTH_PARAMETER_MSG));
+		}
+
+		try
+		{
+			ObjectInputStream objectInputStream =
+				new ObjectInputStream(new FileInputStream(filename));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. The resource is referenced at lines 123, 124. The resource is closed at line 125. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).